scala/private/rule_impls.bzl

```diff
@@ -481,21 +480,27 @@ def is_dependency_analyzer_off(ctx):
 # Extract very common code out from dependency analysis into single place
 # automatically adds dependency on scala-library and scala-reflect
 # collects jars from deps, runtime jars from runtime_deps, and
-def _collect_jars_from_common_ctx(ctx, extra_deps = [],
+def _collect_jars_from_common_ctx(ctx,
+                                  default_compile_classpath,
+                                  default_runtime_classpath,
+                                  extra_deps = [],
                                   extra_runtime_deps = []):
 
   dependency_analyzer_is_off = is_dependency_analyzer_off(ctx)
 
-  # Get jars from deps
-  scalac_provider = ctx.attr._scalac[_ScalacProvider]
-  auto_deps = [scalac_provider.scalalib, scalac_provider.scalareflect]
-  deps_jars = collect_jars(ctx.attr.deps + auto_deps + extra_deps,
+  default_cjars = [d[JavaInfo].compile_jars for d in default_compile_classpath]
+  default_transitive_rjars = [
+      d[JavaInfo].transitive_runtime_jars for d in default_runtime_classpath
+  ]
+
+  deps_jars = collect_jars(ctx.attr.deps + extra_deps,
                            dependency_analyzer_is_off)
-  (cjars, transitive_rjars, jars2labels,
-   transitive_compile_jars) = (deps_jars.compile_jars,
-                               deps_jars.transitive_runtime_jars,
-                               deps_jars.jars2labels,
-                               deps_jars.transitive_compile_jars)
+
+  (cjars, transitive_rjars, jars2labels, transitive_compile_jars) = (
+      depset(transitive = [deps_jars.compile_jars] + default_cjars),
```

Comment:
I think you're changing the semantics here because now one can get a warning about the usage of the default compile classpath. Might be wrong but worth taking another look

Reply:
I'll change the naming to `base_compile_classpath`, let me know if there are some bigger changes in the semantics that you'd like